### ISSUE TITLE
Stop using XComGameState_Effect components

### ISSUE THREAD

**Issue by Favid** [Favid/LongWar2AbilitiesForWotC#49]
Abilities using an XComGameState_Effect component can cause hard to reproduce crashes due to them not unregistering from events properly at the end of a tactical mission. Below are steps to reproduce this issue with LW2WotC_LightningReflexes:

1. Give a soldier Lightning Reflexes
2. Take that soldier on a mission and complete it
3. Start a new mission.

The game will crash when starting a new mission, presumably because there are still listeners tied to gamestates that shouldn't be. Or something, I don't totally understand it yet, but creating a new event that is registered for 'TacticalGameEnd' with a callback function that unregisters all events fixes the crash.

Long story short, we need to stop using XComGameState effects and use X2EventListerners instead. If that is not possible for some abilities for some reason, then at least make sure that events get unregistered at the end of a tactical mission.

This will require re-implementing:
- Hyper-Reactive Pupils
- Lightning Reflexes
- Locked On
- Rapid Reaction
- Savior
- Sentinel
- Steady Weapon
- Trojan
- Failsafe
- Bombardier

The following abilities should be fine, but should be heavily retested to make sure they don't cause similar issues:
- Close Encounters
- Emergency Life Support
- Hit and Run

**Comment by Favid** [comment]
Steady Weapon is fine, probably because it is not an effect with an infinite duration. Close Encounters, Emergency Life Support, and Hit and Run are all fine. The others cause the crash and will need to be reworked.

**Comment by Favid** [comment]
Going to consider this issue complete. The only ability that hasn't been rewritten is Steady Weapon, but I've spent a lot of time trying to get it to break the game and it just hasn't. If a bug with it is found in the future, it'll get its own issue.
